src/main/java/Naega.java

```diff
@@ -0,0 +1,150 @@
+                    handleMark(tasks, userInput);
+                } else if (userInput.startsWith("unmark ")) {
+                    handleUnmark(tasks, userInput);
+                } else if (userInput.startsWith("todo ")) {
+                    handleTodo(tasks, userInput);
+                } else if (userInput.startsWith("deadline ")) {
+                    handleDeadline(tasks, userInput);
+                } else if (userInput.startsWith("event ")) {
+                    handleEvent(tasks, userInput);
+                } else if (userInput.startsWith("delete ")) {
+                    handleDelete(tasks, userInput);
+                } else {
+                    throw new NaegaException("I don't know how to handle that command.");
+                }
+            } catch (NaegaException e) {
+                System.out.println("____________________________________________________________");
+                System.out.println(" Oops! " + e.getMessage());
+                System.out.println("____________________________________________________________");
+            } catch (NumberFormatException | IndexOutOfBoundsException e) {
+                System.out.println("____________________________________________________________");
+                System.out.println(" Oops! Invalid task number.");
+                System.out.println("____________________________________________________________");
+            }
+        }
+
+        scanner.close();
+    }
+
+    private static void handleList(ArrayList<Task> tasks) {
+        System.out.println("____________________________________________________________");
+        System.out.println(" Here are the tasks in your list:");
+        for (int i = 0; i < tasks.size(); i++) {
+            System.out.println(" " + (i + 1) + "." + tasks.get(i));
+        }
+        System.out.println("____________________________________________________________");
+    }
+
+    private static void handleMark(ArrayList<Task> tasks, String userInput) throws NaegaException {
+        int taskIndex = parseTaskIndex(userInput, "mark ");
+        tasks.get(taskIndex).markAsDone();
+        System.out.println("____________________________________________________________");
+        System.out.println(" Nice! I've marked this task as done:");
+        System.out.println("   " + tasks.get(taskIndex));
+        System.out.println("____________________________________________________________");
+    }
+
+    private static void handleUnmark(ArrayList<Task> tasks, String userInput) throws NaegaException {
+        int taskIndex = parseTaskIndex(userInput, "unmark ");
+        tasks.get(taskIndex).markAsNotDone();
+        System.out.println("____________________________________________________________");
+        System.out.println(" OK, I've marked this task as not done yet:");
+        System.out.println("   " + tasks.get(taskIndex));
+        System.out.println("____________________________________________________________");
```

Comment:
I like how these methods were named. Good job!

Perhaps common code could be extracted out tho.